src/main/java/seedu/address/storage/RoomOccupancyStorage.java

```diff
@@ -45,18 +54,57 @@ public RoomBook readOnlyRoomOccupancy() throws IOException {
                 rooms.add(room);
                 roomsInArray[i] = room;
             }
-            while (roomsOccupiedFile.hasNextLine()) {
-                int roomNumber = roomsOccupiedFile.nextInt(); //room number that is currently occupied
+            for (int i = 0; i < numberOfRoomsOccupied; i++) {
+                String roomNum = roomsOccupiedFile.next();
+                int roomNumber = Integer.parseInt(roomNum); //room number that is currently occupied
                 Room room = roomsInArray[roomNumber - 1];
                 rooms.remove(room);
```

Comment:
i think you might want to change your naming for the priority queue and not use ``rooms``? i got quite confused trying to read it